### PR TITLE
Move Manage Connections to the Connected Mode menu

### PR DESCRIPTION
**Pull request by rita-sonarsource**
Fixes [#3374](https://github.com/SonarSource/sonarlint-visualstudio/issues/3374)

image

### CODE REVIEW

File: src/Integration.Vsix/Commands/ConnectedModeMenu/ManageConnectionsCommand.cs [hunk 1]
```diff
@@ -22,10 +22,12 @@
 using Microsoft.VisualStudio.Shell;
 using SonarLint.VisualStudio.Integration.TeamExplorer;
 
-namespace SonarLint.VisualStudio.Integration.Vsix
+namespace SonarLint.VisualStudio.Integration.Vsix.Commands.ConnectedModeMenu
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/Commands/ConnectedModeMenu/ManageConnectionsCommand.cs:25]: Moved as-is to the new folder

File: src/Integration.Vsix/Commands/ConnectedModeMenu/ManageConnectionsCommand.cs [hunk 2]
```diff
@@ -22,10 +22,12 @@
...
 {
     internal class ManageConnectionsCommand : VsCommandBase
     {
+        internal const int Id = 0x102;
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/Commands/ConnectedModeMenu/ManageConnectionsCommand.cs:29]: I think it makes more sense to just call it Id, this way the usage is `ManageConnectionsCommand.Id` and not `ManageConnectionsCommand.ManageConnectionsCommandId`

File: src/Integration.Vsix/Commands/PackageCommandId.cs [hunk 3]
```diff
@@ -23,7 +23,6 @@ namespace SonarLint.VisualStudio.Integration.Vsix
     internal enum PackageCommandId
     {
         // Buttons
-        ManageConnections = 0x100,
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/Commands/PackageCommandId.cs:26]: Replaced with an Id on the command

File: src/Integration.Vsix/Commands/PackageCommandManager.cs [hunk 4]
```diff
@@ -64,6 +63,10 @@ public PackageCommandManager(IMenuCommandService menuService)
             this.RegisterCommand(CommonGuids.HelpMenuCommandSet, (int)PackageCommandId.SonarLintHelpShowLogs, new ShowLogsCommand(outputWindowService));
             this.RegisterCommand(CommonGuids.HelpMenuCommandSet, ViewDocumentationCommand.ViewDocumentationCommandId, new ViewDocumentationCommand(showInBrowserService));
             this.RegisterCommand(CommonGuids.HelpMenuCommandSet, AboutCommand.AboutCommandId, new AboutCommand(browserService));
+
+            // Connected mode buttons
+            this.RegisterCommand(CommonGuids.ConnectedModeMenuCommandSet, ManageConnectionsCommand.Id, new ManageConnectionsCommand(teamExplorerController));
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/Commands/PackageCommandManager.cs:68]: Same command but different placement and id

File: src/Integration.Vsix/SonarLintDaemon/DaemonCommands.vsct [hunk 5]
```diff
@@ -42,6 +42,11 @@
       <IDSymbol name="grpDaemonErrorList" value="0x100" />
       <IDSymbol name="cmdidCreateCFamilyReproducer" value="0x300" />
     </GuidSymbol>
+
+    <!-- VS Code Analysis menu/group Guid:IDs -->
+    <GuidSymbol name="guidCodeAnalysisMenuGroup" value="{FBC1A65B-A334-4EC8-8FBE-DB1C56335FB6}">
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/SonarLintDaemon/DaemonCommands.vsct:47]: This was in the PackageCommands.vsct but used also by a command in this file: https://github.com/SonarSource/sonarlint-visualstudio/blob/7f57da532ea85d27ec9072bfaa010f09fc7801cd/src/Integration.Vsix/SonarLintDaemon/DaemonCommands.vsct#L27

It's not needed there anymore so I moved it here

File: src/Integration.Vsix/Commands/ConnectedModeMenu/ConnectedModeMenu.vsct [hunk 6]
```diff
@@ -8,11 +8,32 @@
       </Group>
     </Groups>
 
+    <Buttons>
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/Commands/ConnectedModeMenu/ConnectedModeMenu.vsct:11]: As agreed with Marco, the new command is only visible in the SonarLint menu, so there is no need for command placement like the original command had

File: src/Integration.Vsix/Commands/ConnectedModeMenu/ConnectedModeMenu.vsct [hunk 7]
```diff
@@ -8,11 +8,32 @@
...
+      <Button guid="ConnectedModeMenuCmdSet" id="ManageServerConnectionsCommand">
+        <Parent guid="ConnectedModeMenuCmdSet" id="ConnectedModeGroup"/>
+        <Icon guid="guidIcons" id="bmpSonarQube"/>
+        <Strings>
+          <ButtonText>Manage Server Connections...</ButtonText>
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/Commands/ConnectedModeMenu/ConnectedModeMenu.vsct:16]: Exact string TBD

File: src/Integration.Vsix/Commands/ConnectedModeMenu/ManageConnectionsCommand.cs (hunk 2 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/Integration.Vsix/Commands/ConnectedModeMenu/ManageConnectionsCommand.cs:29]: It did it the same name, so +1

File: src/Integration.Vsix/Commands/ConnectedModeMenu/ConnectedModeMenu.vsct [hunk 8]
```diff
@@ -8,11 +8,32 @@
       </Group>
     </Groups>
 
+    <Buttons>
+      <Button guid="ConnectedModeMenuCmdSet" id="ManageServerConnectionsCommand">
+        <Parent guid="ConnectedModeMenuCmdSet" id="ConnectedModeGroup"/>
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration.Vsix/Commands/ConnectedModeMenu/ConnectedModeMenu.vsct:13]: maybe place it into a different group, so it's distinct from the issue viz buttons?

Comment (rita-sonarsource) [reply src/Integration.Vsix/Commands/ConnectedModeMenu/ConnectedModeMenu.vsct:13]: At the moment it's distinct thanks to the fact they _they're_ in a different group :)

Comment (georgii-borovinskikh-sonarsource) [reply src/Integration.Vsix/Commands/ConnectedModeMenu/ConnectedModeMenu.vsct:13]: yeah ok, then it's all good

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**